executor/inspection_summary.go

```diff
@@ -415,10 +415,23 @@ func (e *inspectionSummaryRetriever) retrieve(ctx context.Context, sctx sessionc
 		return nil, nil
 	}
 	e.retrieved = true
-
 	rules := inspectionFilter{set: e.extractor.Rules}
 	names := inspectionFilter{set: e.extractor.MetricNames}
 
+	// Get clusterInfo from global cache.
+	// If InstanceAddrCache is nil, set it and release it in the end.
+	var clusterInfo map[string]string
+	if sctx.GetSessionVars().InstanceAddrCache == nil {
+		var err error
+		clusterInfo, err = GetInstanceMap(sctx)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		clusterInfo = sctx.GetSessionVars().InstanceAddrCache
+	}
+	defer func() { sctx.GetSessionVars().InstanceAddrCache = nil }()
```

Comment:
I think we should assign `sctx.GetSessionVars().InstanceAddrCache = clusterInfo` at L429 and use the `defer` function following this assignment to clear the cache.